src/js/components/sharedComponents/timeChart/chart/BarChartLegendItem.jsx

```diff
@@ -3,36 +3,49 @@
  * Created by Kevin Li 3/21/17
  */
 
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
+import { throttle } from "lodash";
+import { smTabletScreen } from "../../../../dataMapping/shared/mobileBreakpoints";
 
 const propTypes = {
     color: PropTypes.string,
     label: PropTypes.string,
     offset: PropTypes.number
 };
 
-export default class BarChartLegendItem extends React.Component {
-    render() {
-        return (
-            <g
-                className="chart-legend-item"
-                transform={`translate(${this.props.offset}, 0)`}>
-                <circle
-                    className="key-color"
-                    fill={this.props.color}
-                    cx="6"
-                    cy="6"
-                    r="6" />
-                <text
-                    className="key-label"
-                    x="20"
-                    y="10">
-                    {this.props.label}
-                </text>
-            </g>
-        );
-    }
-}
+const BarChartLegendItem = (props) => {
+    const [windowWidth, setWindowWidth] = useState(0);
+
+    useEffect(() => {
+        const handleResize = throttle(() => {
+            const newWidth = window.innerWidth;
+            if (windowWidth !== newWidth) {
+                setWindowWidth(newWidth);
+            }
+        }, 50);
+        window.addEventListener('resize', handleResize);
+        return () => window.removeEventListener('resize', handleResize);
+    }, [windowWidth]);
+    return (
```

Comment:
it would be great to have a newline here, between the useEffect and the return function